src/lib/context.go

```diff
@@ -73,10 +76,12 @@ func WithArtifactInfo(ctx context.Context, art ArtifactInfo) context.Context {
 }
 
 // GetArtifactInfo gets the ArtifactInfo from the context
-func GetArtifactInfo(ctx context.Context) (art ArtifactInfo) {
+func GetArtifactInfo(ctx context.Context) (art ArtifactInfo, err error) {
```

Comment:
I don' t think an object does not exist in context should be an error.

Reply:
you mean, we do check none in all of the callers, and define whether to return 404?